kernel/nfft/nfft.c

```diff
@@ -2478,6 +2478,12 @@ static void nfft_adjoint_1d_B(X(plan) *ths)
 
 void X(trafo_1d)(X(plan) *ths)
 {
+  if((ths->N[0] <= ths->m) || (ths->n[0] <= 2*ths->m+2))
+  {
+    X(trafo_direct)(ths);
```

Comment:
Longer term, we should use a pointer to the correct trafo method in the plan. But that's a topic for NFFT 4.0. I suggest we create a branch where we can work on the change towards an opaque plan pointer. For some time, we will still need to support the 3.x.x line, but we should get started on the larger re-design.